### PR TITLE
MGDOBR-1331 Use uber-jar for smart events deliverables

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-1331](https://issues.redhat.com/browse/MGDOBR-1331)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by r00ta — COMMENTED** [review]
Sorry, forget it

**Review by vaibhavjainwiz — APPROVED** [review]
lgtm

File: processor/processor-configuration/src/main/java/com/redhat/service/smartevents/processor/ProcessorCatalogServiceImpl.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ public List<String> getSourcePasswordProperties(String id) {
     }
 
     private String readFile(String resourceDirectory, String name) {
-        InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(resourceDirectory + name + JSON_FILE_EXTENSION);
+        InputStream is = this.getClass().getResourceAsStream(resourceDirectory + name + JSON_FILE_EXTENSION);
```
Comment (robpblake) [inline processor/processor-configuration/src/main/java/com/redhat/service/smartevents/processor/ProcessorCatalogServiceImpl.java:119]: @sutaakar Out of interest, why the need for this change?

**Review by robpblake — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

File: processor/processor-configuration/src/main/java/com/redhat/service/smartevents/processor/ProcessorCatalogServiceImpl.java (hunk 1 above)
Comment (sutaakar) [reply processor/processor-configuration/src/main/java/com/redhat/service/smartevents/processor/ProcessorCatalogServiceImpl.java:119]: @robpblake The ContextClassLoader wasn't able to resolve the resource from uber-jar, I guess it doesn't have access to the main jar, probably just to its dependencies.
System ClassLoader can resolve resources from uber-jar.

**Pull request merged**